text/204-golang-bindings.md

```diff
@@ -0,0 +1,669 @@
+        startPeriod: hc.StartPeriod(),
+        timeout:     hc.Timeout(),
+    }
+}
+```
+
+#### Case 2: Extending Typescript datatype interfaces
+
+For Typescript datatype interfaces that extend another datatype interface, the corresponding interfaces and implementing structs would be generated in
+go, with the extending interface embedding the extended one, e.g.:
+
+```ts
+export interface BaseServiceOptions {
+  readonly cluster: ICluster;
+  readonly desiredCount?: number;
+  readonly serviceName?: string;
+  readonly maxHealthyPercent?: number;
+  readonly minHealthyPercent?: number;
+  readonly healthCheckGracePeriod?: Duration;
+  readonly cloudMapOptions?: CloudMapOptions;
+  readonly propagateTags?: PropagatedTagSource;
+  readonly enableECSManagedTags?: boolean;
+  readonly deploymentController?: DeploymentController;
+}
+
+export interface BaseServiceProps extends BaseServiceOptions {
+  readonly launchType: LaunchType;
+}
+```
+
+As before, the Go translation would have the getter methods (e.g. readonly cluster) defined in a the generated interface, which correspond to
+non-exported struct members in the struct that implements each getter.
+
+```go
+package ecs
+
+type BaseServiceOptionsIface interface {
+    Cluster()                ICluster
+    DesiredCount()           *int
+    ServiceName()            *string
+    MaxHealthyPercent()      *int
+    MinHealthyPercent()      *int
+    HealthCheckGracePeriod() *Duration
+    CloudMapOptions()        CloudMapOptions
+    PropagateTags()          PropagatedTagSource
+    EnableECSManagedTags()   *bool
+    DeploymentController()   DeploymentController
+}
+
+type BaseServicePropsIface interface {
+    BaseServiceOptionsIface  // embeddeded interface
+    LaunchType()             LaunchType
+}
+
+type BaseServiceOptions struct {
+    cluster                ICluster
+    desiredCount           *int
+    serviceName            *string
+    maxHealthyPercent      *int
+    minHealthyPercent      *int
+    healthCheckGracePeriod *Duration
+    cloudMapOptions        CloudMapOptions
+    propagateTags          PropagatedTagSource
+    enableECSManagedTags   *bool
+    deploymentController   DeploymentController
+}
+
+type BaseServiceProps struct {
+    BaseServiceOptions  // embedded (anonymous) field
+    launchType          LaunchType
+}
+
+func (o BaseServiceOptions) Cluster() ICluster { return o.cluster }
+func (o BaseServiceOptions) DesiredCount() int { return o.desiredCount }
+
+// ... etc
+
+```
+
+This allows the embedding interface (here, `BaseServicePropsIFace`) to "inherit" all the methods defined in the embedded interface
+(`BaseServiceOptionsIFace`). Similarly, the embedding struct (`BaseServiceProps`) would have access to all the properties of the embedded struct
+(`BaseServiceOptions`) through having it as an anonymous field (See: [Go playground example](https://play.golang.org/p/XRpTH3LN6bx)).
+
+```go
+props := BaseServiceProps{
+    BaseServiceOptions{
```

Comment:
Since it's hard to keep track of comments once the PR is updated, Just noting that Elad had a similar concern [here](https://github.com/aws/aws-cdk-rfcs/pull/206#discussion_r456901209)

and Jason also discusses it [here](https://github.com/aws/aws-cdk-rfcs/pull/206#discussion_r458262827)